x-pack/docs/en/security/authentication/saml-guide.asciidoc

```diff
@@ -322,7 +322,15 @@ groups:: _(Recommended)_
     If you wish to use your IdP's concept of groups or roles as the basis for a
     user's {es} privileges, you should map them with this attribute.
     The `groups` are passed directly to your
-    <<saml-role-mapping, role mapping rules>>
+    <<saml-role-mapping, role mapping rules>>.
++
+[NOTE]
+====
+Some IdPs are only able to send the `groups` list as a comma-separated string.
+{es} can't parse this string into an array of groups. However, you can use a
+wildcard when <<saml-role-mapping,configuring role mappings>> to map this SAML
+attribute to the `attributes.groups` setting in the {es} realm.
+====
```

Comment:
Here is some more information that may help paraphrase these sentences a bit more: It's mostly likely a configuration issue rather than idPs "not able to send". That said, sometimes our customres have no control over their idPs. In those situations, the end result is the same for them, i.e. they recieve a comma-separate string that is meant to be a list. ES does not (for now) attempt to split it. 

It is recommended to fix it on the idP side. But as said above, customers may not have control. For now the workaround is to use wildcard (regex is more accurate but we haven't really document its general usage other than [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/defining-roles.html#CO545-5)).

Reply:
Thanks for the context @ywangd! Per our conversation, we can also add a suggestion about regex here with an example from #84379. Let me know what you think.

> Some IdPs are configured to send the `groups` list as a comma-separated string, but {es} can't parse this string into an array of groups. To map this SAML attribute to the `attributes.groups` setting in the {es} realm, a cluster security administrator can use a wildcard when <<saml-role-mapping,configuring role mappings>>. Alternatively, a cluster security administrator can use a regular expression to create a role mapping rule such as `/^(.*,)?elasticsearch-admins(,.*)?$/` that matches on multiple groups. These regular expressions are based on Lucene’s regexp automaton syntax, and can match more complex patterns. All regular expressions must start and end with a forward slash.

Reply:
Thanks Adam.
* We can link to the doc page of [lucene regex syntax](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html)
* "Alternatively ... matches on _multiple_ groups" - It actually only matches a single group which is `elasticsearch-admins`. The extra stuff around it is to make sure this name can appear anywhere in the comma separate list, e.g. it can be in the beginning where it is _not_ preceded with a comma; or it can be at the end of the string where it is _not_ followed by a comma; or it can be in the middle where it is both preceded and followed by a comma; or it can be standalone where it is the only member of the string. Using only wildcard is not possible to achieve this level of accuracy and flexibility. For example, `*elasticsearch-admins*` can also match `my-elasticsearch-admins` which is probably not what the user wants. Or `*,elasticsearch-admin,*`, but it wont' match if `elasticsearch-admins` is at the begining of the string.

Reply:
Thanks Yang. I updated the note in [this recent push](https://github.com/elastic/elasticsearch/pull/86770/commits/fc46081becdd582ad56701e7c0edf97611aa0d6d).